staging/src/k8s.io/cloud-provider/app/controllermanager.go

```diff
@@ -294,6 +294,7 @@ func startControllers(cloud cloudprovider.Interface, ctx genericcontrollermanage
 	}
 
 	c.SharedInformers.Start(stopCh)
+	ctx.InformerFactory.Start(ctx.Stop)
```

Comment:
Really need to rename the controllercontext to not confuse it with a regular context. Not your issue though.

Reply:
Would be nice if we had a test which demonstrated this was necessary.

Reply:
Any pointers on this? I know that this is required in practice because the nodeipam controller does not start up correctly when run in cloud-controller-manager without this fix. Any way to simulate this in the test? I can take a look at the tests to try this.

Reply:
Currently we don't have support to start CCM in k/k which might block adding tests. -_-

Reply:
That blocks a full e2e test. I wonder if we could have unit test which starts a generic controller manager unit test and verifies that the InformerFactory is running.